Add git/github workflow for making changes when you don't own a repository for hacktoberfest (#65)
Arcadia-Science/arcadia-computational-training: docs/workshops/20231006-github-hacktoberfest/lesson.md

# Using Git and GitHub to Contribute to Repositories You Don't Have Write Access To

Arcadia is participating in [Hacktoberfest 2023](https://hacktoberfest.com/).
Our goal is to give back to the open source software community that we use every day in our own scientific work.
As part of this, we are hosting a mini-hackathon to create a space for and support others in making contribution to open source software libraries.
We've [curated a list of bioinformatics (or adjacent) software repos](https://docs.google.com/document/d/1EjyQOwPO-Zc4aKM_mTPEFo3K97Ox3lEk-DyZxlivyVU/edit?usp=sharing) that are accepting contributions.

> The purpose the lesson below is to outline how to contribute to a GitHub repository that you don't own (e.g., you don't have write access to).

Reviewer: ```suggestion
The purpose of the lesson below is to outline how to contribute to a GitHub repository that you don't own (e.g., you don't have write access to).
```

4. Make your changes.
5. Add your changes to be tracked by git: `git add file-changed.txt`
6. Commit your changes: `git commit -m 'short explanation of changes i made'`
7. Push your changes: `git push origin branch-name`
8. Create a new pull request from your forked repository (click the Pull request banner and then the New Pull Request button located at the top of your repo).
9. Describe your changes following the guidelines in the contribution guide or the PR template.
10. Wait for your PR to be reviewed and make any requested changes.

## Setting up

If you need to:

- sign up for a GitHub account
- install git locally
- configure global git settings
- create an ssh key to authenticate without a password

Navigate to the [Setting Up Instructions](https://training.arcadiascience.com/workshops/20220920-intro-to-git-and-github/lesson/#setting-up) in the Intro to Git and GitHub lesson and follow along with the tutorial.

## Finding repositories that accept contributions & work that needs to be done

### Searching for repositories that are open to external contributions

Not every open source repository is in a position to accept work from new contributors.

> Contributing to a repository that isn't accepting these types of contributions at best leads to unintegrated pull requests and at worse creates additional work and head aches for maintainers.

Reviewer: ```suggestion
Contributing to a repository that isn't accepting these types of contributions at best leads to unintegrated pull requests and at worse creates additional work and headache for maintainers.
```

You can start from this list or search GitHub topics to find a repository you want to contribute to.

For the hands on portions in the rest of this lesson, we will practice making a contribution using the [Arcadia-Science/2023-hackathon-practice](https://github.com/Arcadia-Science/2023-hackathon-practice) repository.

### Contributor guidelines 

The Contributor Guidelines for a repository outline how the project would like to receive contributions.
This is a very important document to read and to continually reference as you're making your contribution to make sure you're following the preferences of the community or project you're contributing to.
Each project is a little different, so make sure you read this document closely for each project you plan to contribute to.

In the [Arcadia-Science/2023-hackathon-practice](https://github.com/Arcadia-Science/2023-hackathon-practice) repository, the contributor guide is located in the file `CONTRIBUTING.md` and it is linked to from the `README.md`.

### Finding work that needs to be completed: issues, pull requests, and repo searches

> The contributor guidlines document often details how new contributors can find work that needs to be done.

Reviewer: ```suggestion
The contributor guidelines document often details how new contributors can find work that needs to be done.
```
